save-backend/src/main/kotlin/com/saveourtool/save/backend/security/ProjectPermissionEvaluator.kt

```diff
@@ -88,11 +86,17 @@ class ProjectPermissionEvaluator {
 
     private fun Authentication.hasRole(role: Role): Boolean = authorities.any { it.authority == role.asSpringSecurityRole() }
 
-    private fun hasWriteAccess(userId: Long?, project: Project, projectRole: Role): Boolean = when {
-        userId == null -> false
-        project.userId == userId -> true
-        else -> projectRole == Role.ADMIN || projectRole == Role.OWNER
-    }
+    private fun hasReadAccess(userId: Long?, projectRole: Role): Boolean = userId?.let {
+        projectRole == Role.VIEWER || projectRole == Role.ADMIN || projectRole == Role.OWNER || projectRole == Role.SUPER_ADMIN
+    } ?: false
+
+    private fun hasWriteAccess(userId: Long?, projectRole: Role): Boolean = userId?.let {
```

Comment:
maybe refactor to reuse previous function to show that one access extends another?
```suggestion
    private fun hasWriteAccess(userId: Long?, projectRole: Role): Boolean = hasDeleteAccess(userId, projectRole) || userId?. let { projectRole == Role.ADMIN } ?: false
```

Reply:
Makes sense, thx